tensorflow/stream_executor/blas.h

```diff
@@ -1383,6 +1455,93 @@ class BlasSupport {
                           const DeviceMemory<std::complex<double>> &a, int lda,
                           DeviceMemory<std::complex<double>> *b, int ldb) = 0;
 
+  // Creates a backend-specific plan object for a blaslt matmul operation, which
+  // can then be passed to DoBlasLtMatmul(). When possible, plans should be
+  // created once and reused for multiple calls to DoBlasLtMatmul().
+  // Returns a null pointer on failure.
+  std::unique_ptr<blas::IBlasLtMatmulPlan> CreateBlasLtMatmulPlan(
+      blas::DataType ab_type, blas::DataType c_type,
+      blas::ComputationType computation_type, blas::PointerMode pointer_mode,
+      blas::Transpose transa, blas::Transpose transb, uint64 m, uint64 n,
+      uint64 k, int64 lda, int64 ldb, int64 ldc) {
+    return CreateBlasLtMatmulPlanStridedBatched(
+        ab_type, c_type, computation_type, pointer_mode, transa, transb, m, n,
+        k, 1, lda, 0, ldb, 0, ldc, 0);
+  }
+
+  // A more general version of CreateBlasLtMatmulPlan supporting
+  // batched operations.
+  virtual std::unique_ptr<blas::IBlasLtMatmulPlan>
+  CreateBlasLtMatmulPlanStridedBatched(
+      blas::DataType ab_type, blas::DataType c_type,
+      blas::ComputationType computation_type, blas::PointerMode pointer_mode,
+      blas::Transpose transa, blas::Transpose transb, uint64 m, uint64 n,
+      uint64 k, int batch_count, int64 lda, int64 stride_a, int64 ldb,
+      int64 stride_b, int64 ldc, int64 stride_c) = 0;
+
+  // Gets a list of supported algorithms for DoBlasLtMatmul. The algorithms are
+  // returned in the order of increasing estimated compute time according to an
+  // internal heuristic. The first returned algorithm can be used as the default
+  // algorithm if no autotuning is to be performed.
+  virtual bool GetBlasLtMatmulAlgorithms(
+      const blas::IBlasLtMatmulPlan* plan, size_t max_workspace_size,
+      int max_algorithm_count,
+      std::vector<std::unique_ptr<blas::IBlasLtMatmulAlgorithm>>*
+          out_algorithms) = 0;
+
+  // Executes a blaslt matmul operation on the stream. If output_profile_result
+  // is not nullptr, the operation is profiled, error messages are
+  // suppressed, and output_profile_result->algorithm() is set to
+  // algorithm->index().
+  virtual bool DoBlasLtMatmul(
```

Comment:
Are we able to remove all these overloadings by similarly using a struct for parameters?

Reply:
Removed by using runtime types instead of static type overloads.